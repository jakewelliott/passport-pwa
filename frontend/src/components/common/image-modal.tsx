import React from 'react';
import { TransformWrapper, TransformComponent } from "react-zoom-pan-pinch";

interface ImageModalProps {
  photo: {
    url: string;
    caption?: string;
  };
  onClose: () => void;
}

export const ImageModal: React.FC<ImageModalProps> = ({ photo, onClose }) => {
  return (
    <div className="fixed inset-0 bg-system_black flex items-center justify-center" style={{zIndex: 9999}}>
      <span className="absolute top-4 right-6 text-system_white text-h1 font-bold cursor-pointer z-10" onClick={onClose}>&times;</span>
      <TransformWrapper>
        <TransformComponent>
          <img 
            src={photo.url} 
            alt={photo.caption || 'Photo'} 
            className="max-w-full max-h-screen object-contain"
          />
        </TransformComponent>
      </TransformWrapper>
      {photo.caption && (
        <div className="absolute bottom-4 left-0 right-0 text-center text-system_white bg-system_black bg-opacity-50 p-2">
          {photo.caption}
        </div>
      )}
    </div>
  );
};
