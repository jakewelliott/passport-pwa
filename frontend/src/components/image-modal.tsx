import { a11yOnClick } from '@/lib/a11y';
import type React from 'react';
import { TransformComponent, TransformWrapper } from 'react-zoom-pan-pinch';

interface ImageModalProps {
    photo: {
        photoPath: string;
        alt?: string;
    };
    onClose: () => void;
}

export const ImageModal: React.FC<ImageModalProps> = ({ photo, onClose }) => {
    return (
        <dialog
            aria-modal='true'
            className='fixed inset-0 flex items-center justify-center bg-system_black'
            style={{ zIndex: 9999 }}
        >
            <span
                className='absolute top-4 right-6 z-10 cursor-pointer font-bold text-h1 text-system_white'
                style={{ textShadow: '0 0 4px #444444' }}
                {...a11yOnClick(onClose)}
            >
                &times;
            </span>
            <TransformWrapper>
                <TransformComponent>
                    <img
                        src={photo.photoPath}
                        alt={photo.alt || 'Photo'}
                        className='max-h-screen max-w-full object-contain'
                    />
                </TransformComponent>
            </TransformWrapper>
            {photo.alt && (
                <div
                    data-testid='caption-container'
                    className='absolute right-0 bottom-4 left-0 bg-system_black bg-opacity-50 p-2 text-center text-system_white'
                >
                    {photo.alt}
                </div>
            )}
        </dialog>
    );
};
