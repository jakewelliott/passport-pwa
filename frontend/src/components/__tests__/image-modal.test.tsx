import { fireEvent, render, screen } from '@testing-library/react';
import { beforeEach, describe, expect, it, vi } from 'vitest';
import { ImageModal } from '../image-modal';

// Mock react-zoom-pan-pinch as it's an external dependency
vi.mock('react-zoom-pan-pinch', () => ({
    TransformWrapper: ({ children }: { children: React.ReactNode }) => (
        <div data-testid='transform-wrapper'>{children}</div>
    ),
    TransformComponent: ({ children }: { children: React.ReactNode }) => (
        <div data-testid='transform-component'>{children}</div>
    ),
}));

describe('ImageModal', () => {
    const mockOnClose = vi.fn();
    const mockPhoto = {
        photoPath: 'test-image.jpg',
        alt: 'Test Caption',
    };

    beforeEach(() => {
        mockOnClose.mockClear();
    });

    it('matches snapshot', () => {
        const { container } = render(<ImageModal photo={mockPhoto} onClose={mockOnClose} />);
        expect(container).toMatchSnapshot();
    });

    it('calls onClose when close button is clicked', () => {
        render(<ImageModal photo={mockPhoto} onClose={mockOnClose} />);

        const closeButton = screen.getByText('Ã—');
        fireEvent.click(closeButton);
        expect(mockOnClose).toHaveBeenCalledTimes(1);
    });

    it('renders with zoom wrapper components', () => {
        render(<ImageModal photo={mockPhoto} onClose={mockOnClose} />);

        expect(screen.getByTestId('transform-wrapper')).toBeInTheDocument();
        expect(screen.getByTestId('transform-component')).toBeInTheDocument();
    });
});
