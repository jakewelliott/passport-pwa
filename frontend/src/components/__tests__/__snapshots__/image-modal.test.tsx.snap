// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ImageModal > matches snapshot 1`] = `
<div>
  <div
    class="fixed inset-0 flex items-center justify-center"
    style="z-index: 9999;"
  >
    <div
      class="absolute inset-0 bg-system-black bg-opacity-70 backdrop-blur-md backdrop-filter"
    />
    <dialog
      aria-modal="true"
      class="relative flex h-full w-full items-center justify-center bg-system-black bg-opacity-0"
    >
      <span
        class="absolute top-4 right-6 z-10 cursor-pointer font-bold text-h1 text-system-white"
        style="text-shadow: 0 0 4px #444444;"
      >
        Ã—
      </span>
      <div
        data-testid="transform-wrapper"
      >
        <div
          data-testid="transform-component"
        >
          <div
            class="flex h-full w-full items-center justify-center"
          >
            <img
              alt="Test Caption"
              class="object-contain"
              src="test-image.jpg"
              style="width: 0px; height: 0px; max-width: 100vw; max-height: 100vh;"
            />
          </div>
        </div>
      </div>
      <div
        class="absolute right-0 bottom-4 left-0 bg-system-black bg-opacity-50 p-2 text-center text-system-white"
        data-testid="caption-container"
      >
        Test Caption
      </div>
    </dialog>
  </div>
</div>
`;
